packages/stream/sui/StreamClient.ts

```diff
@@ -85,33 +86,50 @@ export default class SuiStreamClient extends BaseStreamClient {
     { senderWallet }: ICreateStreamSuiExt
   ): Promise<IMultiTransactionResult> {
     const wallet = new SuiWalletWrapper(senderWallet, this.client);
-    const transactions = await this.generateCreateTransactions(wallet.address, multipleStreamData);
+    const [txb, firstIndex] = await this.generateCreateBlock(wallet.address, multipleStreamData);
 
     const txs: string[] = [];
     const metadatas: string[] = [];
     const metadataToRecipient: Record<string, IRecipient> = {};
     const errors: ICreateMultiError[] = [];
 
-    for (let i = 0; i < transactions.length; i++) {
-      const txb = transactions[i];
-      const recipient = multipleStreamData.recipients[i];
-      try {
-        const { digest, events } = await wallet.signAndExecuteTransactionBlock({
-          transactionBlock: txb,
-          options: { showEvents: true },
+    try {
+      const { digest, events, effects } = await wallet.signAndExecuteTransactionBlock({
+        transactionBlock: txb,
+        options: { showEffects: true, showEvents: true },
+      });
+      txs.push(digest);
+
+      if (effects!.status.status === "failure") {
+        multipleStreamData.recipients.forEach((recipient) => {
+          errors.push({
+            error: effects!.status.error ?? "Unknown error!",
+            recipient: recipient.recipient,
```

Comment:
Looks great! I guess there is no way to execute successful ones and return failed?

Reply:
Thanks!